### PR TITLE
For sequencing add log api to get a list of configured logs

### PR DESCRIPTION
**Pull request by Martin2112**
Allow option to limit the results to logs that have unsequenced leaves to be processed to avoid then having to do status queries on each log.

### CODE REVIEW

File: storage/log_storage.go [hunk 1]
```diff
@@ -82,3 +83,11 @@ type LogRootWriter interface {
 	// StoreSignedLogRoot stores a freshly created SignedLogRoot.
 	StoreSignedLogRoot(root trillian.SignedLogRoot) error
 }
+
+// LogMetadata provides access to information about the logs in storage
+type LogMetadata interface {
+	// GetActiveLogs returns the IDs of all the logs that are configured in storage
+	// if filterPendingWorkOnly is true then the list of logs returned includes only those
+	// with pending leaves that need to be integrated into the log.
+	GetActiveLogIDs(filterPendingWorkOnly bool) ([]trillian.LogID, error)
```
Comment (AlCutter) [inline storage/log_storage.go]: I wonder if this wouldn't just be better as two distinct API calls - they each seem to stand on their own as necessary functionality.

Comment (Martin2112) [inline storage/log_storage.go]: Can do.

**Comment by Martin2112** [comment]
Done.

File: storage/mysql/log_storage.go [hunk 2]
```diff
@@ -558,11 +558,12 @@ func (t* logTX) getActiveLogIDsInternal(sql string) ([]trillian.LogID, error) {
...
-	if (filterPendingWorkOnly) {
-		return t.getActiveLogIDsInternal(selectActiveLogsWithUnsequencedSql)
-	}
-
+// GetActiveLogIDs returns a list of the IDs of all configured logs
+func (t *logTX) GetActiveLogIDs() ([]trillian.LogID, error) {
 	return t.getActiveLogIDsInternal(selectActiveLogsSql)
 }
+
+// GetActiveLogIDsWithPendingWork returns a list of the IDs of all configured logs
```
Comment (AlCutter) [inline storage/mysql/log_storage.go]: "... with pending work"?

File: storage/mysql/storage_test.go [hunk 3]
```diff
@@ -819,7 +819,7 @@ func TestGetActiveLogIDsFiltered(t *testing.T) {
 	s := prepareTestStorage(logID, t)
 	tx := beginTx(s, t)
 
-	logIDs, err := tx.GetActiveLogIDs(true)
+	logIDs, err := tx.GetActiveLogIDsWithPendingWork()
```
Comment (AlCutter) [inline storage/mysql/storage_test.go]: Probably worth renaming these tests too?

File: storage/mysql/log_storage.go (hunk 2 above)
Comment (Martin2112) [inline storage/mysql/log_storage.go]: Ooops following line went missing. Reinstated.

File: storage/mysql/storage_test.go (hunk 3 above)
Comment (Martin2112) [inline storage/mysql/storage_test.go]: Done, and fixed incorrect test setup in both.

**Pull request merged**